numpy/core/src/private/ufunc_override.c

```diff
@@ -119,12 +121,26 @@ PyUFunc_WithOverride(PyObject *args, PyObject *kwds,
             }
         }
         /*
-         * Now see if the object provides an __array_ufunc__. However, we should
-         * ignore the base ndarray.__ufunc__, so we skip any ndarray as well as
-         * any ndarray subclass instances that did not override __array_ufunc__.
+         * Have we seen this class before?  If so, ignore.
          */
-        method = get_non_default_array_ufunc(obj);
-        if (method != NULL) {
+        if (with_override != NULL) {
```

Comment:
What about when `with_override == NULL`? Won't things get called twice? When is it NULL?